docs/faq/ImmutableData.md

```diff
@@ -1,245 +1,145 @@
+## What approaches are there for handling data immutably? Do I have to use Immutable.js? 
 
-```
-import { connect } from 'react-redux';
+You do not need to use Immutable.JS with Redux. Plain JavaScript, if written correctly, is perfectly capable of providing immutability without having to use an immutable-focused library. However, guaranteeing immutability with JavaScript is difficult, and it can be easy to mutate an object accidentally, causing bugs in your app that are extremely difficult to locate. For this reason, using a library such as Immutable.JS can significantly improve the reliability of your app, and make your app’s development much easier.
 
-import { toJS } from `./to-js';
 
-import DumbComponent from './dumb.component';
+<a id="issues-with-es6-for-immutable-ops"></a>
+## What are the issues with using plain JavaScript for immutable operations?
+JavaScript was never designed to provide guaranteed immutable operations. Accordingly, there are several issues you need to be aware of if you choose to use it for your immutable operations in your Redux app.
 
-const mapStateToProps = (state) => {
-   return {
-      /**
-      obj is an Immutable object in Smart Component, but it’s converted to a plain 
-      JavaScript object by toJS, and so passed to DumbComponent as a pure JavaScript 
-      object. Because it’s still an Immutable.JS object here in mapStateToProps, though, 
-      there is no issue with errant re-renderings.
-      */
-       obj: getImmutableObjectFromStateTree(state)
-   }
-};
+#### Accidental Object Mutation
+With JavaScript, you can accidentally mutate an object (such as the Redux state tree) quite easily without realising it. For example, updating deeply nested properties, creating a new *reference* to an object instead of a new object, or performing a shallow copy rather than a deep copy, can all lead to inadvertent object mutations, and can trip up even the most experienced JavaScript coder. 
 
-export default connect(mapStateToProps)(toJS(DumbComponent));
-```
-By converting Immutable.JS objects to plain JavaScript values within a HOC, we achieve Dumb Component portability, but without the performance hits of using `toJS()` in the Smart Component.
+To avoid these issues, ensure you follow the recommended [immutable update patterns for ES6](http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html).
 
-#### Use the Immutable Object Formatter Chrome Extension to Aid Debugging
+#### Verbose Code
+Updating complex nested state trees can lead to verbose code that is tedious to write and difficult to debug.
 
-Install the [Immutable Object Formatter](https://chrome.google.com/webstore/detail/immutablejs-object-format/hgldghadipiblonfkkicmgcbbijnpeog) , and inspect your Immutable.JS data without seeing the noise of Immutable.JS's own object properties.
+#### Poor Performance
+Operating on JavaScript objects and arrays in an immutable way can be slow, particularly as your state tree grows larger.
```

Comment:
Should clarify why/how perf is an issue - as far as I know, it's primarily a problem with copying keys from an old item to a new item.  Great article on the topic at https://medium.com/@dtinth/immutable-js-persistent-data-structures-and-structural-sharing-6d163fbd73d2 that's worth linking here.